argilla-server/src/argilla_server/validators/records.py

```diff
@@ -108,6 +109,36 @@ def _validate_image_field(self, field_name: str, field_value: Union[str, None])
         else:
             raise UnprocessableEntityError(f"image field {field_name!r} has an invalid URL value")
 
+    def _validate_chat_fields(self, dataset: Dataset, fields: Dict[str, str]) -> None:
+        for field in filter(lambda field: field.is_chat, dataset.fields):
+            self._validate_chat_field(field.name, fields.get(field.name))
+
+    def _validate_chat_field(self, field_name: str, field_value: Union[str, None]) -> None:
+        if field_value is None:
+            return
+
+        if len(field_value) > 5000:
+            raise UnprocessableEntityError(
+                f"chat field {field_name!r} value is exceeding the maximum length of 5000 characters"
+            )
+
+        if not isinstance(field_value, list):
+            raise UnprocessableEntityError(f"chat field {field_name!r} value must be a list of dictionaries")
+
+        for i, value in enumerate(field_value):
```

Comment:
I think we should also validate if we don't pass anything extra.

Reply:
Sorry, I don't quite get this. We should validate that field_value does contains only content and role keys?

Reply:
@burtenshaw what happens if we pass.
```
{
    "role": "user",
    "content": "Hello!",
    "extra": "Hello!"
 }
```